### DISCUSSION TITLE
Serenity IIS Deployment Publish

### DISCUSSION THREAD

**Discussion by havvanur160401046** [serenity-is/Serenity#5315] [Help]
I have completed my Serenity project, I publish to visual studio and then I get an error after I install IIS.
![image](https://user-images.githubusercontent.com/24252615/102755811-fbd6d480-437f-11eb-8b0b-ad6cc86ab6a0.png)

What steps should I do until I deploy my project to IIS? Can you help me?

**Comment by JohnRanger** [comment]
Hi @havvanur160401046 

If you read the error message you get you will clearly see that your asp.net app (your serenity app) on your IIS machine is not able to connect to the necessary serenity databases on your SQL server. 

So either you are not using a real Microsoft SQL server for your deployment (e.g. using only the Visual Studio-builtin DEV SQL server for developing and you do not migrate your MS SQL databases from your DEV machine onto a real MS SQL server - or you did the migration and there is a network problem (e.g. a firewall) between your IIS machine and the SQL server where the Serenity Databases of your project are running on.

If it is only a small database (will never grow over 10 GB) then you could also use MS SQL Express Version which is free.

Please note that express versions (at least in the past) didn't by default allow DB access from a remote machine - it only allowed access from localhost (meaning, that the IIS service and the SQL service must be installed on the same server). In order to allow remote access to the SQL server service, you have to configure the SQL instance to allow for remote database access. Google is your friend in that matter.

Hope, this helps you.

With kind regards,

John
